src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -1071,45 +1070,37 @@ static AnalysisCacheData()
+                // cache file name.
+                int index = 0;
+                string[] featureNames = new string[ExperimentalFeature.EnabledExperimentalFeatureNames.Count];
+                foreach (string featureName in ExperimentalFeature.EnabledExperimentalFeatureNames)
                 {
-                    // If any experimental features are enabled, we cannot use the default cache file because those
-                    // features may expose commands that are not available in a regular powershell session, and we
-                    // should not cache those commands in the default cache file because that will result in wrong
-                    // auto-completion suggestions when the default cache file is used in another powershell session.
-                    //
-                    // Here we will generate a cache file name that represent the combination of enabled feature names.
-                    // We first convert enabled feature names to lower case, then we sort the feature names, and then
-                    // compute an SHA1 hash from the sorted feature names. We will use a short SHA name (first 8 chars)
-                    // to generate the cache file name.
-                    int index = 0;
-                    string[] featureNames = new string[ExperimentalFeature.EnabledExperimentalFeatureNames.Count];
-                    foreach (string featureName in ExperimentalFeature.EnabledExperimentalFeatureNames)
-                    {
-                        featureNames[index++] = featureName.ToLowerInvariant();
-                    }
-
-                    Array.Sort(featureNames);
-                    string allNames = string.Join(Environment.NewLine, featureNames);
+                    featureNames[index++] = featureName.ToLowerInvariant();
+                }
 
-                    // Use SHA1 because it's faster.
-                    // It's very unlikely to get collision from hashing the combinations of enabled features names.
-                    hashBytes = sha1.ComputeHash(Encoding.UTF8.GetBytes(allNames));
+                Array.Sort(featureNames);
+                string allNames = string.Join(Environment.NewLine, featureNames);
 
-                    // Use the first 8 characters of the hash string for a short SHA.
-                    hashString = BitConverter.ToString(hashBytes, startIndex: 0, length: 4).Replace("-", string.Empty);
-                    cacheFileName = string.Format(CultureInfo.InvariantCulture, "{0}-{1}", cacheFileName, hashString);
-                }
+                // Use CRC32 because it's faster.
+                // It's very unlikely to get collision from hashing the combinations of enabled features names.
+                hashString = CRC32Hash.ComputeHash(allNames);
+                cacheFileName = string.Format(CultureInfo.InvariantCulture, "{0}-{1}", cacheFileName, hashString);
```

Comment:
Why not `GetHashCode`, or if that's not stable across runs, the a simple non-allocating hash function.

Reply:
`GetHashCode` in .NET Core 2.1 returns different values from different processes for the same string. That's a by design change (Marvin algorithm) in .NET Core to prevent DOS attack.

Reply:
But you can have a simpler stable hash function, e.g. this [one](https://github.com/PowerShell/PowerShell/blob/b0daa1e724cbba5e00f118001f1105bb44d91952/src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs#L1804)

Reply:
There as some comments about the choice of `31` as the prime number in the algorithm
https://stackoverflow.com/questions/299304/why-does-javas-hashcode-in-string-use-31-as-a-multiplier

Reply:
Interesting, I'm more than happy to replace CRC32 with a simpler stable hash function. Will take a look.

Reply:
Seems I know how to make the simple function faster. :-) https://github.com/dotnet/corefx/pull/32949#discussion_r236468915

Reply:
Update: https://source.dot.net/#System.Private.CoreLib/shared/System/Globalization/TextInfo.cs,609

Reply:
Thanks @iSazonov. Wait on the feedback from other folks. Once we land on the simple hash function, I can make the change in a separate follow-up PR.

Reply:
I opened #8364 to track the follow-up work for myself. I will fix it in a separate PR.

Reply:
Although the hash function is called is one-time for the module analysis, it's called at startup every time.  The [MurmurHash3](https://github.com/aappleby/smhasher) might be a better choice for speed and uniqueness (avoiding collisions), but we can defer that to outside this PR if we open an issue.

Reply:
I think you should prefer the simplest possible code for this use case. The actual hash function isn't too important, it's the side effects (like loading an assembly, memory allocation) that I think you want to avoid.

Reply:
Thanks for all the discussion here. I will take all input into account when working on the hash function PR.